src/Utils/request/useQuery.ts

```diff
@@ -42,6 +43,16 @@ export default function useQuery<TData>(
     }
   }, [runQuery, options?.prefetch]);
 
+  useEffect(() => {
+    if (options?.refetchOnUpdate) {
+      runQuery();
+
+      // window.addEventListener("focus", onFocus);
+
+      // return () => window.removeEventListener("focus", onFocus);
+    }
+  }, [runQuery, options?.refetchOnUpdate]);
+
```

Comment:
You need not add this, `useQuery` will automatically ALWAYS do a refetch whenever the body/query or anything from the options changes and is intended.

Reply:
Whenever the `options` changes, `runQuery` is updated since runQuery is from `useCallback` with `options` as it's dependency.

And whenever `runQuery` is updated, and `prefetch` is true, it'll refetch.

All u need to do is ensure prefetch is not set to `false` when this is supposed to happen.

Reply:
If you need to force `refetch`, you can always deconstruct the `refetch` from the return type of the `useQuery` and invoke that too.

Reply:
@rithviknishad got it, updated the PR